crates/script/src/lib.rs

```diff
@@ -588,6 +570,24 @@ impl ScriptConfig {
             .gas_limit(self.evm_opts.gas_limit());
 
         if let Some(script_wallets) = script_wallets {
+            let mut inner = script_wallets.inner.lock();
+            // if a sender is provided it will be set on evm_opts already
+            // otherwise if we have exactly one signer we can assume it is the sender
+            // else we throw an error to avoid landing on the DEFAULT_SENDER.
+            if inner.provided_sender.is_none() {
+                // may unlock some pending keystores
+                let signers = inner.multi_wallet.signers()?;
+                if signers.len() == 1 {
+                    let address = signers.keys().next().unwrap();
+                    self.evm_opts.sender = *address;
+                }
+                if signers.len() > 1 {
+                    eyre::bail!("multiple signers found, please provide a --sender address.");
```

Comment:
cc @tchardin 

Looks like requires some additional changes due to failing tests:

- forge::cli script::can_broadcast_script_skipping_simulation
- forge::cli script::can_deploy_library_create2_different_sender